### PR TITLE
adder api and documentation cleanup

### PR DESCRIPTION
**Pull request by desmonddak**
## Description & Motivation

Converted all adders to use the same base Adder class with a conditional carry input (Logic? carryIn).

## Related Issue(s)

None

## Testing

Most adders are now tested uniformly in the adder_test.dart file using both carry and non-carry configurations.

## Backwards-compatibility

> Is this a breaking change that will not be backwards-compatible? If yes, how so?

Yes.  The FullAdder no longer has a carryOut, it is in sum[1].  Otherwise, other adders now have an optional carryIn.

## Documentation

> Does the change require any updates to documentation? If so, where? Are they included?

Yes.  All adder documentation has been updated to reflect the change.

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -52,14 +55,17 @@ Some in-development items will have opened issues, as well. Feel free to create
...
-    - Double (64-bit)
-    - Float (32-bit)
-    - BFloat16 (16-bit)
-    - BFloat8 (8-bit)
-    - BFloat4 (4-bit)
+    - [Floating-Point Value Types](./components/floating_point.md#floatingpointvalue)
+      - Double (64-bit)
+      - Float (32-bit)
+      - BFloat16 (16-bit)
+      - TensorFloat32 (32-bit)
```
Comment (mkorbel1) [inline doc/README.md:62]: tf32 is 19 bits?

File: doc/components/adder.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Adder
+# Adders
 
-ROHD-HCL provides a set of adder modules to get the sum from a pair of Logic. As of now, we have
+ROHD-HCL provides a set of adder modules to get the sum from a pair of Logic. Adders which provide a carry input have a base class of [AdderCi](https://intel.github.io/rohd-hcl/rohd_hcl/Adder-class.html), those that don't use a base class of [Adder](https://intel.github.io/rohd-hcl/rohd_hcl/Adder-class.html). As of now, ROHD-HCL supplies:
```
Comment (mkorbel1) [inline doc/components/adder.md:3]: typo: "AdderCi"?

File: doc/components/adder.md [hunk 3]
```diff
@@ -48,16 +48,16 @@ Here is an example of instantiating a `ParallelPrefixAdder`:
     print('${sum.value.toBigInt()}');
 ```
 
-## One's Complement Adder Subtractor
+## Ones Complement Adder Subtractor
```
Comment (mkorbel1) [inline doc/components/adder.md:51]: why remove apostrophe?
https://en.wikipedia.org/wiki/Two%27s_complement
https://en.wikipedia.org/wiki/Ones%27_complement

**Review by mkorbel1 — CHANGES_REQUESTED** [review]
Looks great, thank you!

Just a few minor things

File: doc/README.md (hunk 1 above)
Comment (desmonddak) [reply doc/README.md:62]: Yes.  I didn't realize and fixed this.  Good catch!

File: doc/components/adder.md (hunk 2 above)
Comment (desmonddak) [reply doc/components/adder.md:3]: Forgot to save all from VSCode.  Fixed.

File: doc/components/adder.md (hunk 3 above)
Comment (desmonddak) [reply doc/components/adder.md:51]: I am moving to using ones' and twos' -complement throughout.

**Review by mkorbel1 — APPROVED** [review]
Looks great!

**Pull request merged**